### PR TITLE
fix: to_json renders NaN and Infinity as JSON strings

### PR DESCRIPTION
**Pull request by davidlghellin**
- Render NaN, Infinity, and -Infinity as JSON strings ("NaN", "Infinity", "-Infinity") instead of Rust's inf/-inf forms
- Add BDD feature tests for special float values, empty structures, arrays, nested maps, and null handling

### CODE REVIEW

File: python/pysail/tests/spark/function/features/to_json.feature [hunk 1]
```diff
@@ -1,3 +1,4 @@
+@to_json
```
[the remark this answers is not in the document]
Comment (linhr) [reply python/pysail/tests/spark/function/features/to_json.feature]: Yes this marker needs to be removed.
